src/frontend/apps/lti_site/types/metadata.ts

```diff
@@ -1,16 +1,20 @@
 import { RouteOptions } from './RouteOptions';
-import { Video } from 'lib-components';
+import { TimedText, Video } from 'lib-components';
 
 export interface ResourceMetadata {
   name: string;
   description: string;
   renders: string[];
   parses: string[];
-  actions: RouteOptions<Video>['actions'];
 }
 
 export interface VideoMetadata extends ResourceMetadata {
   live: {
     segment_duration_seconds: number;
   };
+  actions: RouteOptions<Video>['actions'];
+}
+
+export interface TimedTextMetadata extends ResourceMetadata {
+  actions: RouteOptions<TimedText>['actions'];
 }
```

Comment:
And what do you think about this ?

```
import { RouteOptions } from './RouteOptions';
import { Resource, TimedText, Video } from 'lib-components';

export interface ResourceMetadata<T extends Resource> {
  name: string;
  description: string;
  renders: string[];
  parses: string[];
  actions: RouteOptions<T>['actions'];
}

export interface VideoMetadata extends ResourceMetadata<Video> {
  live: {
    segment_duration_seconds: number;
  };
  
}

export interface TimedTextMetadata extends ResourceMetadata<TimedText> {}
```

Reply:
Yes better, I didn't have the right syntax in mind